demo-app/src/main.rs

```diff
@@ -0,0 +1,253 @@
+        sovereign_sdk::stf::ConsensusSetUpdate<OpaqueAddress>,
+    > {
+        let mut storage = self.current_storage.clone();
+        let mut events = Vec::new();
+        for tx in batch.take_transactions() {
+            // Do mock signature verification
+            // We just check that the signature hash matches the tx hash
+            let expected_hash = tx.mock_signature.msg_hash;
+            let found_hash = <MockContext as Spec>::Hasher::hash(&tx.msg);
+            // If the (mock) signature is invalid, the sequencer is malicious. Slash them.
+            if expected_hash != found_hash {
+                return Err(ConsensusSetUpdate::slashing(sequencer));
+            }
+
+            if let Ok(msg) = RuntimeCall::<MockContext>::decode(&mut std::io::Cursor::new(tx.msg)) {
+                let ctx = MockContext::new(tx.mock_signature.pub_key);
+                let tx_result = msg.dispatch_call(storage.clone(), &ctx);
+                match tx_result {
+                    Ok(resp) => {
+                        events.push(resp.events);
+                        storage.merge();
+                    }
+                    Err(_) => {
+                        storage.merge_reads_and_discard_writes();
+                    }
+                }
+            } else {
+                // If the serialization is invalid, the sequencer is malicious. Slash them
+                return Err(ConsensusSetUpdate::slashing(sequencer));
+            }
+        }
+        Ok(events)
+    }
+
+    fn apply_proof(
+        &self,
+        _proof: Self::Proof,
+        _prover: &[u8],
+    ) -> Result<(), sovereign_sdk::stf::ConsensusSetUpdate<OpaqueAddress>> {
+        todo!()
+    }
+
+    fn end_slot(
+        &mut self,
+    ) -> (
+        Self::StateRoot,
+        Vec<sovereign_sdk::stf::ConsensusSetUpdate<OpaqueAddress>>,
+    ) {
+        self.current_storage.finalize();
+        todo!()
+    }
+}
+
+#[derive(Debug, PartialEq, BorshDeserialize, BorshSerialize)]
+pub struct Batch<C: Context> {
+    txs: Vec<DemoTransaction<C>>,
+}
+
+impl<C: Context> BatchTrait for Batch<C> {
+    type Transaction = DemoTransaction<C>;
+
+    fn transactions(&self) -> &[Self::Transaction] {
+        &self.txs
+    }
+
+    fn take_transactions(self) -> Vec<Self::Transaction> {
+        self.txs
+    }
+}
+
+#[derive(Genesis, DispatchCall, DispatchQuery, MessageCodec)]
+pub struct Runtime<C: Context> {
+    #[allow(unused)]
+    election: Election<C>,
+}
+
+impl<C: Context> core::fmt::Debug for Runtime<C> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Runtime").finish()
+    }
+}
+
+impl<C: Context> TransactionTrait for DemoTransaction<C> {
+    type Hash = [u8; 32];
+}
+
+impl<C: Context> CanonicalHash for DemoTransaction<C> {
+    type Output = [u8; 32];
+
+    fn hash(&self) -> Self::Output {
+        self.mock_signature.msg_hash.clone()
+    }
+}
+
+fn main() {
+    type RT = Runtime<MockContext>;
```

Comment:
Let's move the content of the `basic_runtime ` here and delete `sov-modules/sov-modules-impl/integration-tests/tests/demo` So the integration test becomes the `demo app`.  

`basic_runtime`  defines the `execute` function which we can keep for this PR but the intention is to replace it with `apply_batch` and the code there is structured to make this change easy.

Reply:
Removed the existing `main` body.